server/src/domain/auth/auth.service.ts

```diff
@@ -337,7 +337,7 @@ export class AuthService {
     try {
       const issuer = await Issuer.discover(issuerUrl);
       const algorithms = (issuer.id_token_signing_alg_values_supported || []) as string[];
-      if (algorithms[0] === 'HS256') {
+      if (algorithms[0] === config.oauth.signingAlgorithm) {
```

Comment:
If it is set use it, otherwise don't set it. (No need to compare to [0] anymore

Reply:
Oh but what if it's not supported by the issuer?

Reply:
Doesn't matter. If the user put it in just use it as is. The user is specifically saying that the auth server has been configured this way for this client.